tfhe/src/high_level_api/keys/server.rs

```diff
@@ -152,12 +155,14 @@ impl Versionize for ServerKey {
             integer_key: self.key.versionize(),
         })
     }
+}
 
+impl VersionizeOwned for ServerKey {
     type VersionedOwned = ServerKeyVersionedOwned;
 
-    fn versionize_owned(&self) -> Self::VersionedOwned {
+    fn versionize_owned(self) -> Self::VersionedOwned {
         ServerKeyVersionedOwned::V0(ServerKeyVersionOwned {
-            integer_key: self.key.versionize_owned(),
+            integer_key: (*self.key).clone().versionize_owned(),
```

Comment:
I'm confused here, why the clone ?

Reply:
key is wrapped in an Arc here. So I have to extract the inner type to call `versionize_owned`. MSRV for the tfhe crate is 1.75 so I cannot use this: https://doc.rust-lang.org/std/sync/struct.Arc.html#method.unwrap_or_clone

Reply:
However I should at least `impl<T: Versionize> Versionize for Arc<T>` in `tfhe-versionable` I think

Reply:
you are allowed to bump the version and you can do the impl if you want to as well

Reply:
if you don't need the MSRV bump then don't bump

Reply:
I added the impl for Arc<T>, this simplifies the code in `tfhe` since the manual impl is not needed anymore.
I also updated the MSRV.